lib/Sema/CSSimplify.cpp

```diff
@@ -3762,21 +3762,42 @@ ConstraintSystem::SolutionKind ConstraintSystem::simplifyMemberConstraint(
                        e.second == MemberLookupResult::UR_TypeMemberOnInstance;
               })) {
 
+        SmallVector<OverloadChoice, 4> choices;
+
+        for (auto &pair : result.UnviableCandidates) {
+          choices.push_back(std::move(pair.first));
+        }
+
+        if (baseTy->is<MetatypeType>() &&
+            baseTy->getMetatypeInstanceType()->isAnyExistentialType()) {
+          if (llvm::all_of(choices, [&](const OverloadChoice &c) {
+                return c.isDecl() && isa<ConstructorDecl>(c.getDecl());
+              })) {
+
+            for (auto choice : choices) {
+              bool isStaticallyDerived =
+                  !(baseTy->getMetatypeInstanceType()->is<DynamicSelfType>() ||
+                    baseTy->getMetatypeInstanceType()->is<ArchetypeType>());
+              auto *protoFix = AllowInvalidInitRef::onProtocolMetatype(
+                  *this, baseTy, dyn_cast<ConstructorDecl>(choice.getDecl()),
+                  isStaticallyDerived, choice.getDecl()->getSourceRange(),
+                  locator);
+              if (recordFix(protoFix))
+                return SolutionKind::Error;
+            }
+
+            addOverloadSet(memberTy, choices, useDC, locator);
+            return SolutionKind::Solved;
+          }
+        }
+
         auto *fix =
             AllowTypeOrInstanceMember::create(*this, baseTy, member, locator);
```

Comment:
Can you try to simply form the overload set without producing this fix if base meets protocol metatype criteria, and choices are constructors (you can check the first one via `llvm::any_of` - if it's `ConstructorDecl` the rest are too). I think it should work just fine because there is `validateInitializerRef` now which is going to inroduce appropriate fixes.

Reply:
@xedin Done!